crates/tests/test-tedge-cli/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+mod test_env;
+
+#[cfg(test)]
+mod tests {
+    use crate::*;
+    use assert_matches::*;
+    use rexpect::errors::*;
+    use rexpect::process::signal::Signal;
+    use rexpect::process::wait::WaitStatus;
+    use rexpect::*;
+
+    macro_rules! execute_cmd {
+        ($($arg:tt)*) => {spawn(&format!($($arg)*), TIMEOUT_MS)?.process.wait()?}
+    }
+
+    macro_rules! spawn_cmd {
+        ($($arg:tt)*) => {spawn(&format!($($arg)*), TIMEOUT_MS)?}
+    }
+
+    const TIMEOUT_MS: Option<u64> = Some(5_000);
+
+    #[test]
+    fn it_works() -> Result<()> {
+        let mut sub = spawn("tedge mqtt sub test/topic", TIMEOUT_MS)?;
+
+        execute(r#"tedge mqtt pub test/topic "hello thin-edge""#)?;
+        assert_eq!(sub.read_line()?, "INFO: Connected");
+        assert_eq!(sub.read_line()?, "[test/topic] hello thin-edge");
+
+        execute(r#"tedge mqtt pub test/topic bye-bye --qos 2"#)?;
+        assert_eq!(sub.read_line()?, "[test/topic] bye-bye");
+
+        sub.process.kill(Signal::SIGTERM)?;
```

Comment:
Note, this test interacts with the command under test (`tedge mqtt pub`) while the command is running.

I think this make easier to understand what is tested even if this is no more following the "Arrange, Act, Assert" pattern.